includes/class-wc-stripe-order-handler.php

```
@@ -522,6 +533,46 @@ public function prevent_cancelling_orders_awaiting_action( $cancel_order, $order
+
+			WC_Stripe_Logger::info( "Settling order {$order->get_id()} from PaymentIntent {$intent_id} ({$intent_status}) instead of cancelling it as unpaid." );
+
+			$order->add_order_note( __( 'Stripe reports this payment as successful, but the payment confirmation never reached this site (e.g. a missed webhook). The order was updated from Stripe instead of being auto-cancelled as unpaid.', 'woocommerce-gateway-stripe' ) );
+
+			$this->process_response( $charge, $order );
+		} catch ( Exception $e ) {
+			WC_Stripe_Logger::error( "Failed to settle order {$order->get_id()} from its PaymentIntent before cancellation: " . $e->getMessage() );
+		} finally {
+			$order_helper->unlock_order_payment( $order );
+		}
+	}
```

PHPMD (2.15.0)

[error] The method settle_paid_order_instead_of_cancelling is not named in camelCase. (undefined)

(CamelCaseMethodName)